catalog/spatial/kml/spatial-kml-transformer/src/main/java/org/codice/ddf/spatial/kml/transformer/KMLTransformerImpl.java

```diff
@@ -208,6 +228,85 @@ protected Placemark performDefaultTransformation(Metacard entry)
     return kmlPlacemark;
   }
 
+  private void setExtendedData(Placemark placemark, Metacard metacard) {
+    final ExtendedData extendedData = new ExtendedData();
+
+    final Set<AttributeDescriptor> attributeDescriptors =
+        metacard.getMetacardType().getAttributeDescriptors();
+
+    for (AttributeDescriptor attributeDescriptor : attributeDescriptors) {
+      final String attributeName = attributeDescriptor.getName();
+      final Attribute attribute = metacard.getAttribute(attributeName);
+      if (attribute != null) {
+        String attributeValue = convertAttribute(attribute, attributeDescriptor).toString();
+        attributeValue = StringUtils.removeStart(StringUtils.removeEnd(attributeValue, "]"), "[");
+        final Data data = getData(attributeName, attributeValue);
+        extendedData.addToData(data);
+      }
+    }
+    placemark.setExtendedData(extendedData);
+  }
+
+  private Data getData(String attributeAlias, String attributeValue) {
+    final Data data = new Data(attributeValue);
+    data.setName(attributeAlias);
+    return data;
+  }
+
+  private Serializable convertAttribute(Attribute attribute, AttributeDescriptor descriptor) {
+
+    if (descriptor.isMultiValued()) {
+      ArrayList<Serializable> values = new ArrayList<>();
+      for (Serializable value : attribute.getValues()) {
+        values.add(
+            convertValue(attribute.getName(), value, descriptor.getType().getAttributeFormat()));
+      }
+      return values;
+    } else {
+      return convertValue(
+          attribute.getName(), attribute.getValue(), descriptor.getType().getAttributeFormat());
+    }
+  }
+
+  private Serializable convertValue(
+      String name, Serializable value, AttributeType.AttributeFormat format) {
+    if (value == null) {
+      return null;
+    }
+
+    switch (format) {
+      case DATE:
+        if (!(value instanceof Date)) {
```

Comment:
Is it possible for the value to not be a date if the attribute is a date type?  I think this would cause a Solr exception on ingest

Reply:
from what i've asked around it doesn't seem possible. very similar code is also being used at https://github.com/codice/ddf/blob/f008f640967fa6a655943f0a8d7068d0c5d063db/catalog/transformer/catalog-transformer-propertyjson-metacard/src/main/java/ddf/catalog/transformer/metacard/propertyjson/PropertyJsonMetacardTransformer.java#L187
hearing more input from others would be helpful on this though